### PR TITLE
Add button to sync Github issue

### PR DESCRIPTION
**Pull request by PierrickGT**
##### Description

As a bounty owner, I wish to be able to update my bounty description by syncing from Github.
This PR adds a Sync Issue button that allows a bounty owner to easily update the bounty description.

JmtsK51Wao

##### Refers/Fixes

Issue: https://github.com/gitcoinco/web/issues/4870

### CODE REVIEW

**Comment by PierrickGT** [comment]
> Sweet! Looks great - lots of people will be stoked for this!

Thank you! I will submit my work tonight once I get home.

**Comment by PierrickGT** [comment]
> > Sweet! Looks great - lots of people will be stoked for this!
> 
> Thank you! I will submit my work tonight once I get home.

I'm gonna need @thelostone-mc to approve me as a worker cause I can't submit my work for now.

**Comment by thelostone-mc** [comment]
Damn neat 🙌 
Ah I'm not sure what happened. I did approve it after we synced up on how you'd tackle this.
Nevertheless -> approved now

File: app/assets/v2/js/pages/bounty_details.js [hunk 1]
```diff
@@ -859,6 +859,57 @@ var show_extend_deadline_modal = function() {
...
+
+            if (result.url) {
+              setTimeout(function() {
+                document.location.href = result.url;
+              }, 1000);
+            }
+          }
+        ).fail(
+          function(result) {
+            var alertMsg = result && result.responseJSON ? result.responseJSON.error : null;
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_details.js]: could we update this to `let` ?

File: app/assets/v2/js/pages/bounty_details.js [hunk 2]
```diff
@@ -859,6 +859,57 @@ var show_extend_deadline_modal = function() {
...
+
+            if (alertMsg === null) {
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_details.js]: thoughts on making it as 

```javascript
let alertMsg = result && result.responseJSON ? result.responseJSON.error : gettext('Failed to sync issue. Please reload the page and try again.');
```

File: app/assets/v2/js/pages/bounty_details.js [hunk 3]
```diff
@@ -859,6 +859,57 @@ var show_extend_deadline_modal = function() {
...
+              alertMsg = gettext('Failed to sync issue. Please reload the page and try again.');
+            }
+            _alert({ message: alertMsg }, 'error');
+          }
+        );
+      }).fail(function(result) {
+        var alertMsg = result && result.responseJSON ? result.responseJSON.error : null;
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_details.js]: ^ `let`

File: app/assets/v2/js/pages/bounty_details.js [hunk 4]
```diff
@@ -859,6 +859,57 @@ var show_extend_deadline_modal = function() {
...
+
+        if (alertMsg === null) {
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_details.js]: let alertMsg = result && result.responseJSON ? result.responseJSON.error : gettext('Failed to sync issue. Please reload the page and try again.');

**Comment by PierrickGT** [comment]
@thelostone-mc I've updated the `alertMsg` syntax: https://github.com/gitcoinco/web/pull/4899/commits/af7d995dcd947931af7cae7ad828e83dd8455731
Let me know what you think.

**Pull request merged**